### ISSUE TITLE
Database upgrade fails with read format error

### ISSUE THREAD

**Issue by Vertganti** [bastillion-io/Bastillion#421]
I am currently attempting to update the [dockerized version of Bastillion by e-cosi](https://github.com/e-COSI/docker-bastillion) to use the most recent version of Bastillion. I included the database upgrade JAR to be run on container start so older databases should be upgraded without requiring any user interaction. However, the attempt to migrate a database with some dummy data from version 3.09.00 to 3.13.00 using `bastillion-upgrade-4.00.01.jar` fails with the following trace:

    connectionURL : jdbc:h2:/opt/bastillion/jetty/bastillion/WEB-INF/classes/file:./keydb/bastillion;CIPHER=AES;
    Downloading https://repo1.maven.org/maven2/com/h2database/h2/1.4.200/h2-1.4.200.jar
    org.h2.jdbc.JdbcSQLNonTransientException: General error: "The read format 1 is smaller than the supported format 2 [2.0.206/5] [50000-206]
            at org.h2.message.DbException.getJdbcSQLException(DbException.java:573)
            at org.h2.message.DbException.getJdbcSQLException(DbException.java:496)
            at org.h2.message.DbException.get(DbException.java:216)
            at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:166)
            at org.h2.mvstore.db.Store.<init>(Store.java:140)
            at org.h2.engine.Database.<init>(Database.java:324)
            at org.h2.engine.Engine.openSession(Engine.java:92)
            at org.h2.engine.Engine.openSession(Engine.java:222)
            at org.h2.engine.Engine.createSession(Engine.java:201)
            at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
            at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:117)
            at org.h2.Driver.connect(Driver.java:59)
            at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
            at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
            at io.bastillion.manage.util.DBUtils.getConn(DBUtils.java:47)
            at Upgrade.main(Upgrade.java:57)
    Caused by: org.h2.mvstore.MVStoreException: The read format 1 is smaller than the supported format 2 [2.0.206/5]
            at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
            at org.h2.mvstore.MVStore.readStoreHeader(MVStore.java:889)
            at org.h2.mvstore.MVStore.<init>(MVStore.java:455)
            at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4052)
            at org.h2.mvstore.db.Store.<init>(Store.java:129)
            ... 11 more
    null

I have looked at #405 which has a similar error message (though there the error was about the `write format`), but it seems that there the error was encountered by Bastillion itself (not the upgrade JAR) and it was resolved by using the upgrade JAR. I hope you can help me resolve this issue without having to resort to exporting and importing the database manually (as suggested in h2database/h2database#3214).

**Comment by skavanagh** [comment]
I just went though it again and I'm not seeing the issue.
```
kavanagh@PC-KAVANAGH:~/Downloads$ tar xfz bastillion-jetty-v3.09_00.tar.gz
kavanagh@PC-KAVANAGH:~/Downloads$ cd Bastillion-jetty/
kavanagh@PC-KAVANAGH:~/Downloads/Bastillion-jetty$ ls
3rdPartyLicenses.md  LICENSE.md  README.md  jetty  startBastillion.bat  startBastillion.sh
kavanagh@PC-KAVANAGH:~/Downloads/Bastillion-jetty$ ./startBastillion.sh
2022-01-17 10:49:14.069:INFO::main: Logging initialized @4229ms to org.eclipse.jetty.util.log.StdErrLog
2022-01-17 10:49:14.400:WARN:oejx.XmlConfiguration:main: Deprecated method public void org.eclipse.jetty.server.HttpConfiguration.setBlockingTimeout(long) in file:///mnt/c/Users/seanp/Downloads/Bastillion-jetty/jetty/etc/jetty.xml
2022-01-17 10:49:14.597:INFO::main: Console stderr/stdout captured to /mnt/c/Users/seanp/Downloads/Bastillion-jetty/jetty/logs/2022_01_17.jetty.log
Please enter database password:
Please confirm database password:

^Ckavanagh@PC-KAVANAGH:~/Downloads/Bastillion-jetty$ ls
3rdPartyLicenses.md  LICENSE.md  README.md  jetty  startBastillion.bat  startBastillion.sh
kavanagh@PC-KAVANAGH:~/Downloads/Bastillion-jetty$ cd ..
kavanagh@PC-KAVANAGH:~/Downloads$ ls
Bastillion-jetty  bastillion-jetty-v3.09_00.tar.gz  bastillion-jetty-v3.13_00.tar.gz  bastillion-upgrade-4.00.01.jar
kavanagh@PC-KAVANAGH:~/Downloads$ mkdir 3.13
kavanagh@PC-KAVANAGH:~/Downloads$ cd 3.13/
kavanagh@PC-KAVANAGH:~/Downloads/3.13$ tar xfz ../bastillion-jetty-v3.13_00.tar.gz
kavanagh@PC-KAVANAGH:~/Downloads/3.13$ cd Bastillion-jetty/jetty/bastillion/WEB-INF/classes/
kavanagh@PC-KAVANAGH:~/Downloads/3.13/Bastillion-jetty/jetty/bastillion/WEB-INF/classes$ cp ~/Downloads/Bastillion-jetty/jetty/bastillion/WEB-INF/classes/
BastillionConfig.properties  bastillion.jceks             jaas.conf                    log4j2.xml
VERSION.txt                  io/                          keydb/
kavanagh@PC-KAVANAGH:~/Downloads/3.13/Bastillion-jetty/jetty/bastillion/WEB-INF/classes$ cp -R ~/Downloads/Bastillion-je
tty/jetty/bastillion/WEB-INF/classes/bastillion.jceks ~/Downloads/Bastillion-jetty/jetty/bastillion/WEB-INF/classes/Bast
illionConfig.properties ~/Downloads/Bastillion-jetty/jetty/bastillion/WEB-INF/classes/keydb .
kavanagh@PC-KAVANAGH:~/Downloads/3.13/Bastillion-jetty/jetty/bastillion/WEB-INF/classes$ java -jar ~/Downloads/bastillion-upgrade-4.00.01.jar /home/kavanagh/Downloads/3.13/Bastillion-jetty/jetty/bastillion/WEB-INF/classes/BastillionConfig.properties
connectionURL : jdbc:h2:/home/kavanagh/Downloads/3.13/Bastillion-jetty/jetty/bastillion/WEB-INF/classes/keydb/bastillion;CIPHER=AES;
conn1: url=jdbc:h2:keydb/bastillion user=BASTILLION
Upgrade successful
kavanagh@PC-KAVANAGH:~/Downloads/3.13/Bastillion-jetty/jetty/bastillion/WEB-INF/classes$ cd ../../../../
kavanagh@PC-KAVANAGH:~/Downloads/3.13/Bastillion-jetty$ ./startBastillion.sh
2022-01-17 10:54:43.232:INFO::main: Logging initialized @3355ms to org.eclipse.jetty.util.log.StdErrLog
2022-01-17 10:54:43.642:INFO::main: Console stderr/stdout captured to /mnt/c/Users/seanp/Downloads/3.13/Bastillion-jetty/jetty/logs/2022_01_17.jetty.log
```

**Comment by skavanagh** [comment]
Let's try this jar to see if we can get some info. There isn't a difference than the one you are running, other than it's printing the stacktraces.  The errors around add columns are to be expected, but maybe we will see something else. 

https://github.com/bastillion-io/Bastillion-upgrade/releases/download/v4.00.01/bastillion-upgrade-4.00.01-verbose.jar

**Comment by Vertganti** [comment]
Thank you for providing the verbose JAR. It revealed that the connection to Maven was failing (due to the Firewall). That error was simply not logged with the default upgrade JAR so it seemed the connection was working. When pursuing a follow-up error I discovered that I had added `file:` to the connection URI, probably due to some bad copypasting when trying to adapt the URL as the author of #417 did. After removing `file:` from the URI the upgrade worked. 

So to sum it up: Two problems on my side, thanks for the help!
